packages/block-editor/src/utils/object.js

```diff
@@ -68,7 +68,8 @@ function cloneObject( object ) {
 		return object.map( cloneObject );
 	}
 
-	if ( object && typeof object === 'object' ) {
+	// Only clone plain objects.
```

Comment:
why?

this function is called `cloneObject()`, so it scares me that it's not deciding to _not_ clone certain things, but I don't understand the function that well.

what if we added a `clone()` method to `RichTextValue` and in this function checked for that?

```js
if ( object && 'function' === typeof object.clone ) {
	return object.clone();
}
```

Reply:
Hehe, I knew you'd comment here. I'm going to look into what's expected here.

Reply:
So turns out this cloning is not necessary, made a preparatory PR here: #56612